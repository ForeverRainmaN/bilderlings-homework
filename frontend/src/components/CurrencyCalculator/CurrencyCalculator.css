.bilderlings-homework-curr-calc {
  flex-direction: column;
}
