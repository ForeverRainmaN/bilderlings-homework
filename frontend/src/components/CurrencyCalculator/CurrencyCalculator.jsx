import React, {PureComponent} from 'react';
import {NumericInput} from '@blueprintjs/core';
import {CurrencySelects} from "../FeeEditor/FeeAddNewForm/FeeAddNewForm";
import {Link} from "react-router-dom";
import './CurrencyCalculator.css';

export default class CurrencyCalculator extends PureComponent {
  constructor(props) {
    const from = ["USD", "EUR", "RUB"];
    const to = ["RUB", "USD", "EUR"];
    super(props);
    this.state = {
      amount: 0.00,
      from: from[0],
      to: to[0]
    };
    this.handleFromChange = this.handleFromChange.bind(this);
    this.handleToChange = this.handleToChange.bind(this);
    this.handleAmountChange = this.handleAmountChange.bind(this);
  }

  handleFromChange({target: {value: from}}) {
    this.setState(() => {
      return {from};
    });
  }

  handleToChange({target: {value: to}}) {
    this.setState(() => {
      return {to};
    });
  }

  handleAmountChange(amount) {
    this.setState(() => {
      return {amount};
    });
  }

  render() {
    const {from, to, amount} = this.state;
    return (
        <div className="bilderlings-homework-curr-calc display-flex">
          <Link to="/fees">Fee Editor</Link>
          <div className="display-flex">
            <NumericInput
                className="bilderlings-homework-amount-input width70-br5-mr-30"
                allowNumericCharactersOnly={true}
                buttonPosition="none"
                onValueChange={this.handleAmountChange}
                value={amount}
            />
            <CurrencySelects
                handleFromChange={this.handleFromChange}
                handleToChange={this.handleToChange}
                toInitialValue={to}
                fromInitialValue={from}
            />
          </div>
        </div>
    );
  }
}
