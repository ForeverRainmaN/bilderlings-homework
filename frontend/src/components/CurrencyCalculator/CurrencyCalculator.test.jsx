import React from "react";
import renderer from "react-test-renderer";
import {MemoryRouter} from "react-router-dom";
import CurrencyCalculator from "./CurrencyCalculator";

it('tests if CurrencyCalculator form renders correctly', () => {
  const addFeeForm = renderer
  .create(
      <MemoryRouter>
        <CurrencyCalculator/>
      </MemoryRouter>).toJSON();
  expect(addFeeForm).toMatchSnapshot();
});
