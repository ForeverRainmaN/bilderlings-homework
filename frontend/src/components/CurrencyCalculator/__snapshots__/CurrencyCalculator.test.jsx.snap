// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests if CurrencyCalculator form renders correctly 1`] = `
<div
  className="bilderlings-homework-curr-calc display-flex"
>
  <a
    href="/fees"
    onClick={[Function]}
  >
    Fee Editor
  </a>
  <div
    className="display-flex"
  >
    <div
      className="bilderlings-homework-amount-input width70-br5-mr-30"
    >
      <div
        className="bp3-input-group"
      >
        <input
          autoComplete="off"
          className="bp3-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onPaste={[Function]}
          style={
            Object {
              "paddingRight": 10,
            }
          }
          type="text"
          value="0"
        />
      </div>
    </div>
    <div
      className="bilderlings-homework-add-fee-form-selects display-flex"
    >
      <select
        className="bliderlings-homework-select-from width70-br5-mr-30"
        defaultValue="USD"
        id="from"
        onChange={[Function]}
      >
        <option>
          USD
        </option>
        <option>
          EUR
        </option>
        <option>
          RUB
        </option>
      </select>
      <select
        className="bliderlings-homework-select-to width70-br5-mr-30"
        defaultValue="RUB"
        id="to"
        onChange={[Function]}
      >
        <option>
          RUB
        </option>
        <option>
          USD
        </option>
        <option>
          EUR
        </option>
      </select>
    </div>
  </div>
</div>
`;
